tiledb/sm/fragment/fragment_info.cc

```diff
@@ -987,6 +978,106 @@ Status FragmentInfo::load_and_replace(
   return Status::Ok();
 }
 
+tuple<optional<Tile>, optional<std::vector<std::pair<std::string, uint64_t>>>>
+load_consolidated_fragment_meta(
+    ContextResources& resources, const URI& uri, const EncryptionKey& enc_key) {
+  auto timer_se =
+      resources.stats().start_timer("sm_read_load_consolidated_frag_meta");
+
+  // No consolidated fragment metadata file
+  if (uri.to_string().empty())
```

Comment:
We can change this to throw and make the return values not optional. The only place it is used dereference the returned tile, which will throw anyways.